assets/javascripts/discourse/initializers/global-filter-preference.js

```diff
@@ -57,7 +57,6 @@ export default {
 
           if (tagFromNewTopic) {
             this._setClientFilterPref(tagFromNewTopic, currentUser);
```

Comment:
Not returning from this condition means that we may pass the condition of `ROUTES_TO_REDIRECT_ON` and have an additional redirect... this makes me nervous. Have you experienced any adverse effects / redirects when testing this on any of the `ROUTES_TO_REDIRECT_ON`?

Reply:
I haven't experienced any adverse effects so far.

the issue is that around 1 out of 8 times I get the wrong app with the `return`, running on `localhost`. On a production instance it's always going to the wrong app